runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6846,6 +6921,37 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
                (TR::Options::getCmdLineOptions()->getOption(TR_EnableJITaaSHeuristics) || localColdCompilations))
                doLocalCompilation = true;
 
+#if DEV_DBG_ENABLE_CHANGE
+            // If it is a remote sync compilation, change it to a local sync compilation.
+            // After the local compilation is completed, a remote async compilation will be scheduled
+            if (!doLocalCompilation &&
+                !entry->_async &&
+                !entry->_changedToLocalSyncComp &&
+                TR::Options::getCmdLineOptions()->getOption(TR_EnableJITaaSHeuristics) &&
+                !TR::CompilationInfo::isCompiled(method) &&
```

Comment:
`TR::CompilationInfo::isCompiled(method)` test should be eliminated. We don't want to do this only for first time sync compilations.